docs/ADR-X-cardano-api-exports-convention.md

```diff
@@ -0,0 +1,43 @@
+# Status
+
+📜 Proposed
+
+# Context
+
+# Proposed Design
+
+## Top level exports
+
+Each top level export should export distinct symbols.
+This may depend on the case, for example: `Cardano.Api.Byron` symbols which are still in use in current eras, can be reexported through `Cardano.Api`.
+
+* `Cardano.Api` - everything domain related for the currently used and upcoming eras
+* `Cardano.Api.E` - everything for the obsoleted `E` era, for example `Cardano.Api.Byron`.
+  1. Any symbol from era `E`, not in use in current era, should be exported through `Cardano.Api.E`.
+  2. Any symbol from era `E` and still in use in current era, should be exported through `Cardano.Api` and additionally through `Cardano.Api.E`.
+    If that's not possible to move it to a module under `Cardano.Api` module, the symbol should be copied to a module under `Cardano.Api`.
+* `Cardano.Api.U` utility module, related to `U` feature set like, like `Cardano.Api.Pretty`, `Cardano.Api.Monad.Error`.
+  Those should **not** be reexported from `Cardano.Api`, because they would pollute symbol namespace, and are not required for using `cardano-api` core features but can be useful for the advanced users.
+  1. Reexports from upstream libraries like consensus, network or ledger, also fall into this category e.g. `Cardano.Api.Ledger`.
+
+  Modules which are exposing general purpose classes and functions should go also into a separate module, and they not need to be reexported through `Cardano.Api`, for example:
```

Comment:
>  they not need to be reexported through `Cardano.Api`

We are in a way re-visiting the problematic question that got us here in the first place. "What should go in Cardano.Api`?

The answer in my opinion is everything. We should group common functionality under a "top level" module (see `Cardano.Api.Serialization`) and simply re-expose all of these "top-level" modules via `Cardano.Api`. For users who only want specific functionality they can pick and choose which modules they ant instead of importing `Cardano.Api`. This also benefits us because we by default export everything via `Cardano.Api`. I think we can avoid name clashes by exporting the utilities separately like you suggested.